db/migrate/20251016163100_add_database_performance_indexes.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class AddDatabasePerformanceIndexes < ActiveRecord::Migration[8.0]
```

Comment:
If you specify `algorithm: :concurrent` and add `disable_ddl_transaction!` to this migration, the indexes can be added without blocking access to the table, ensuring the migration results in no downtime.

There's an example here: https://github.com/nhsuk/manage-vaccinations-in-schools/blob/main/db/migrate/20250916074716_add_index_on_pending_changes_not_empty.rb